os/tools/mkheader.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python
+############################################################################
+#
+# Copyright 2019 Samsung Electronics All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+# either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+#
+############################################################################
+import os
+import sys
+import hashlib
+############################################################################
+# Header information :
+#
+# Header size is 48byte.
+# / header size (2byte) / binary type (2byte) / binary size (4byte)
+# / Binary name (16byte) / Binary version (12byte)
+# / Kernel version (8byte) / Jump address (4byte)
+#
+# Parameter information :
+#
+# Five parameter are required to operate script.
+# argv[1] is file path of binary file.
+# argv[2] is file type.
+# argv[3] is kernel version.
+# argv[4] is binary name.
+# argv[5] is binary version.
+#
+############################################################################
+
+header_size = 48
+file_path  = sys.argv[1]
+binary_type = sys.argv[2]
+kernel_ver = sys.argv[3]
+binary_name = sys.argv[4]
+binary_ver = sys.argv[5]
+
+START = 0
+HEADER_SIZE_S = 2
+BINARY_TYPE_S = 2
+BINARY_SIZE_S = 4
+BINARY_NAME_S = 16
+BINARY_VERSION_S = 12
+KERNEL_VERSION_S = 8
+JUMP_ADDRESS_S = 4
+
+with open(file_path, 'rb') as fp:
+    # binary data copy to 'data'
+    data = fp.read()
+    file_size = fp.tell()
+    fp.close()
+    fp = open(file_path, 'wb')
+
+    # header initialization
+    for i in range(0, header_size):
+        fp.write('\000')
+    OFFSET = START
+
+    fp.seek(OFFSET,0)
+    hex_size = "%04x" %header_size
+    fp.write(hex_size.decode("hex"))
+
+    OFFSET += HEADER_SIZE_S
+    fp.seek(OFFSET,0)
+    hex_type = "%04x" %int(binary_type)
+    fp.write(hex_type.decode("hex"))
+
+    OFFSET += BINARY_TYPE_S
+    fp.seek(OFFSET,0)
+    hex_size = "%08x" %file_size 
+    fp.write(hex_size.decode("hex"))
+
+    OFFSET += BINARY_SIZE_S
+    fp.seek(OFFSET,0)
+    fp.write(binary_name)
+
+    OFFSET += BINARY_NAME_S
+    fp.seek(OFFSET,0)
+    fp.write(binary_ver)
+
+    OFFSET += BINARY_VERSION_S
+    fp.seek(OFFSET,0)
+    fp.write(kernel_ver)
+
+    OFFSET += KERNEL_VERSION_S
+    fp.seek(OFFSET,0)
+    fp.seek(jump_addr_point, 0)
+    
+    # parsing _vector_start address from elf information.
+    # _vector_start is only for ARM architecture. so it
+    # operates in the ARM architecture.
+    os.system('readelf -s ../build/output/bin/tinyara | grep _vector_start > addr_file')
```

Comment:
`../build/output/bin/tinyara` can be changed.
So please define a variable in the top.